generators/entity/index.js

```diff
@@ -1207,6 +1268,34 @@ module.exports = EntityGenerator.extend({
 
     configuring : {
 
+        loadEntityJson: function() {
+            if(!this.importMicroserviceJson) {
+                return;
+            }
+
+            this.fromPath = this.microservicePath + '/' + this.filename;
```

Comment:
Instead of repeating all this again we can ask the question before reading the json file in init stage and just pass the path to be read which would be far simpler